x/incentives/keeper/distribute.go

```diff
@@ -397,11 +421,14 @@ func (k Keeper) Distribute(ctx sdk.Context, gauges []types.Gauge) (sdk.Coins, er
 	return totalDistributedCoins, nil
 }
 
+// Checks if all non perpetual gauges provided have completed their required distributions.
+// If complete, move the gauge from an active to a finished status.
 func (k Keeper) checkFinishDistribution(ctx sdk.Context, gauges []types.Gauge) {
 	for _, gauge := range gauges {
 		// filled epoch is increased in this step and we compare with +1
 		// TODO: Wat? we increment filled epochs earlier, this looks wrong and like
 		// were not paying out the last epoch of rewards...
+		// TODO: @dev explain the above comment further so we can either fix or create issue
```

Comment:
TODO: @ValarDragon can you explain the above comment further so we can either fix or create an issue?